pygmt/figure.py

```diff
@@ -170,6 +170,9 @@ def psconvert(self, **kwargs):
         # Default cropping the figure to True
         if "A" not in kwargs:
             kwargs["A"] = ""
+        # allow for spaces in figure name
+        if kwargs.get("F"):
+            kwargs["F"] = f'"{kwargs.get("F")}"'
```

Comment:
> I think this PR is still needed for people who want to use spaces. It's allowed in GMT (https://docs.generic-mapping-tools.org/dev/begin.html#examples).

Yes, but this is just a patchy solution to the main underlying problem which is #247. I mitigated it in 4126c16160c67f8d2c405e7d5da56145adc989d4 for subplot's `title` parameter too, but I wonder if it's time we have a real, standardized solution that works to fix #247 entirely:

```suggestion
        kwargs["F"] = f'"{kwargs.get("F")}"' if kwargs.get("F") else None
```

Reply:
subprocess provides a function [`list2cmdline`](https://github.com/python/cpython/blob/3.9/Lib/subprocess.py#L533) which may be useful. I think I tried it once. It works but may break backward-compatibility.

Reply:
Oo, that looks promising. How does it work, and why do you think it might break backward-compatibility?

Reply:
I can't find my old testing codes. If I remember it correctly.

Here is the current behavior:
```
# 1st way: FAIL
fig.basemap(R=[0, 100, 0, 50], J="X10c/5c", B='x20+lCrust Age')
# 2nd way: SUCCEED
fig.basemap(R=[0, 100, 0, 50], J="X10c/5c", B='x20+l"Crust Age"')
# 3rd way: FAIL
fig.basemap(R=[0, 100, 0, 50], J="X10c/5c", B="x20+l'Crust Age'")
```
After using `list2cmdline`, all the three options work, but options 2 and 3 will have single or double quotes in the X labels.

Reply:
I see, but maybe we can leave the complicated frame/B option as it is now and only use `list2cmdline` for less complicated parameters, like `fig.text(text=...)` and `fig.savefig(fname=...)`? That itself would be a step in the right direction.

Reply:
We need more investigation on the function and how it can be useful in our project.